formatting_scripts/fix_formatting.sh

```diff
@@ -71,12 +71,14 @@ function run_black_formatting () {
 }
 
 function run_code_spell(){
+    curl https://raw.githubusercontent.com/codespell-project/codespell/v1.14.0/codespell_lib/data/dictionary.txt --output /tmp/dictionary.txt
```

Comment:
what happens if curl fails?

Reply:
hello

Reply:
you would probably just run the script again?

Reply:
This is the error if curl fails, which is confusing. Can we check that curl worked before calling codespell?
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    14  100    14    0     0    168      0 --:--:-- --:--:-- --:--:--   168
Fixing spelling...

Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/codespell", line 558, in <module>
    sys.exit(main(*sys.argv))
  File "/usr/bin/codespell", line 505, in main
    build_dict(options.dictionary)
  File "/usr/bin/codespell", line 284, in build_dict
    [key, data] = line.split('->')
ValueError: not enough values to unpack (expected 2, got 1)
```

Reply:
oof, fixed!